theano/gpuarray/linalg.py

```diff
@@ -377,6 +376,20 @@ def c_lib_dirs(self):
             return [config.magma.library_path]
         return []
 
+    def get_params(self, node):
+        return node.inputs[0].type.context
+
+
+class GpuMagmaSVD(GpuMagmaBase):
+    """Computes the svd of a matrix :math:`A` using magma library.
+    """
+    __props__ = ('full_matrices', 'compute_uv')
+
+    def __init__(self, full_matrices=True, compute_uv=True):
+        self.full_matrices = full_matrices
+        self.compute_uv = compute_uv
+        COp.__init__(self, ['magma_svd.c'], 'APPLY_SPECIFIC(magma_svd)')
+
     def make_node(self, A):
         ctx_name = infer_context_name(A)
         A = as_gpuarray_variable(A, ctx_name)
```

Comment:
As you require A to be C contiguous in C code (magma_svd.c), you should add `A = gpu_contiguous(A)` after this line.